terra-batch-libchart/templates/_helpers.tpl

```diff
@@ -92,14 +92,44 @@ Lookup the existing secret values if they exist, or generate a random value
 {{- end }}
 {{- end }}
 
+{{- define "useLZPostgresWithPgBouncer" -}}
+{{- if and (.Values.postgres.pgbouncer.enabled) (not (.Values.postgres.podLocalDatabaseEnabled)) -}}
+true
+{{- else -}}
+false
+{{- end }}
+{{- end -}}
+
 {{/*
 Return the port to use to connect to postgresql. Allows pgbouncer port to override 
 standard port when appropriate. Non-Azure apps should always use the standard port.
 */}}
 {{- define "postgresPort" -}}
-{{- if and (.Values.postgres.pgbouncer.enabled) (not (.Values.postgres.podLocalDatabaseEnabled)) }}
+{{- if (eq ("true") (include "useLZPostgresWithPgBouncer" .)) }}
 {{- .Values.postgres.pgbouncer.port }}
 {{- else }}
 {{- .Values.postgres.port }}
 {{- end }}
 {{- end -}}
+
+{{/*
```

Comment:
Can you explain the difference between the 3 flags added to this file? I see they're used in different places, but it seems like they are all connected to the same values.

Reply:
This was my effort to make things clearer in all the parts of the chart that use these values. As you point out, they're all connected to the same values, but they're using them to answer different questions. 

On Azure, there are currently three valid database layouts to run this app with:
 * Everything in pod-local Postgres (this is what we have been doing up until now)
 * CBAS and Cromwell in LZ Postgres, TES in pod-local Postgres (this is what we're going to turn on soon)
 * Everything in LZ Postgres (this one is supported by the chart but not TES, so we aren't actually using it yet)
Depending on which of those we're doing, we need to decide whether to deploy Postgres as part of the app or not, and decide how to tell each component to connect to its database. It tuns out to be pretty gross to decide each of those based on the chart values, so I added these flags to make those decisions easier to understand.

 * `useLZPostgresWithPgBouncer` is just used by other functions in this file. It defines the conditions under which we want to run with LZ postgres with PgBouncer at all.
 * `podLocalDatabaseEnabledForCromwellCBAS` is a way to express the above in language that makes more sense with the rest of chart
 * `buildPodLocalDatabase` tells us whether to create the pod-local database in the app.

Reply:
Gotcha, thanks for clarifying!